### DISCUSSION TITLE
Excluded device still reported

### DISCUSSION THREAD

**Discussion by heggink** [zwave-js/zwave-js-ui#1215] [Q&A] [answered]
zw2mqtt version 4.0.1
gateway settings: Use node names off, Ignore location on, include node info on

Building a plugin for domoticz and subscribing to different topics, I noticed that excluded (deleted) devices are still reported through mqtt. Whilst they are no longer in the devices list, they do still list in nodes.json.

Am I doing something wrong or is this intended behaviour? If the last, what is the proper way to prevent excluded nodes from being reported as devices other than manually editing the nodes.json file?

Many thanks!

Herman

**Comment by robertsLando** [comment, marked as answer]
It may be because those topics are retained. Use mqtt exploorer to clean up retained topics once you delete a device :)

**Comment by heggink** [comment]
I just figured that out as well. Thanks for confirming and helping a noob here. I figured out how to delete retained messages and switch off the retained flag so it's all good now.
Many thanks! Great piece of SW. Soooo much better than the original OZW1.6 implementation in domoticz. Finally a decent alternative.

**Reply by robertsLando** [reply]
Thanks ❤️
